### PR TITLE
fix: harden add_environment input validation

### LINKED ISSUE #139: Harden add_environment input validation
## Context

PR #125 added the add-environment UI and WS handler. Copilot flagged a few edge-case hardening items that are low-risk but worth addressing:

## Items

1. **Port range validation** — Port fields accept arbitrary text. Values like -1, 0, or >65535 could produce malformed connection configs. Should use `type="number"` inputs and clamp/validate to [1, 65535], disabling "Add" when invalid.

2. **adapterConfig double-encoding** — The server always `JSON.stringify`s the `adapterConfig` payload. If a client sends an already-serialized string, it gets double-encoded. Should accept both shapes: string as-is, object → stringify, else reject.

3. **ID collision retry loop** — The collision check appends a suffix once. In the extremely unlikely case the suffixed ID also collides, the DB insert throws. Should loop until an unused ID is found or catch and retry.

## Ref

Copilot review comments on PR #125

### PR DESCRIPTION
**Pull request by nick-pape**
## Summary

- **Port range validation**: Changed port inputs from `type="text"` to `type="number"` with `min=1` / `max=65535`. Added `isPortValid()` helper into `isEnvValid()` so the "Add" button is disabled when a port outside `[1, 65535]` is entered (0, negative, or >65535).
- **adapterConfig double-encoding**: The server now accepts a pre-serialized JSON string as-is, stringifies plain objects, and rejects any other type with a descriptive error — preventing double-encoding when a client sends an already-serialized string.
- **ID collision retry loop**: Replaced single-suffix collision check with a bounded loop (up to 10 attempts with suffixed UUIDs, then full UUID fallback) for both `add_environment` and `create_project`.

## Tests added (E2E Playwright)
- Port out-of-range (0, 99999) disables Add button; boundary values (1, 65535) enable it
- `add_environment` WS handler accepts pre-serialized `adapterConfig` string without double-encoding
- `add_environment` WS handler returns error for non-string/non-object `adapterConfig` (e.g. number 42)

Closes #139

### CODE REVIEW

File: packages/server/src/ws-bridge.ts [hunk 1]
```diff
@@ -867,13 +871,26 @@ async function handleMessage(
...
+      if (rawAdapterConfig === undefined || rawAdapterConfig === null) {
+        adapterConfig = "{}";
+      } else if (typeof rawAdapterConfig === "string") {
+        adapterConfig = rawAdapterConfig;
+      } else if (typeof rawAdapterConfig === "object") {
+        adapterConfig = JSON.stringify(rawAdapterConfig);
+      } else {
+        sendWs(ws, { type: "error", payload: { message: "adapterConfig must be an object or JSON string" } });
+        return;
       }
```
[the remark this answers is not in the document]
Comment (nick-pape) [reply packages/server/src/ws-bridge.ts]: Fixed: added `JSON.parse` validation for string adapterConfig inputs. Invalid JSON strings are now rejected, and empty strings are normalized to `{}`.
